### PR TITLE
Post D2 milestone refactor

### PR DESCRIPTION
**Pull request by donljh**

### CODE REVIEW

File: services/question/src/controller/questions-controller.ts [hunk 1]
```diff
@@ -68,30 +68,22 @@ export async function createQuestion(req: Request, res: Response) {
...
     const existingQuestionSnapshot = await db.where('title', '==', data.title).get()
 
-    if (!existingQuestionSnapshot.empty) {
+    if (!existingQuestionSnapshot.empty && existingQuestionSnapshot.docs.some(doc => doc.id !== req.params.id)) {
       return res.status(409).json({ error: 'A question with this title already exists' })
     }
 
-    await db.doc(req.params.id).update(data)
-    res.status(200).json({ message: 'Question updated successfully' })
+    await db.doc(req.params.id).set(data)
```
Comment (bachletuankhai) [inline services/question/src/controller/questions-controller.ts:83]: Should we use `update` only? `set` would overwrite the document, which might delete some properties.

Comment (donljh) [reply services/question/src/controller/questions-controller.ts:83]: `PUT` request semantically replaces the resource. Thought this was the expected behaviour. Also the request payload was changed to become the entire question model (omitting the id), so we can be sure that all the properties in the payload are consistent with the type of the `Question` model

File: frontend/.env.development
Comment (Cloud7050) [inline frontend/.env.development:1]: I find it a bit strange that `frontend\.env.development` is meant to be checked in, but `services\question\.env.development` is not.

**Review by Cloud7050 — APPROVED** [review]

**Comment by Cloud7050** [comment]
Merging this to reattempt PR rebase, as #38 is quite messy rn after merging into it.

File: services/question/src/controller/questions-controller.ts (hunk 1 above)
Comment (ys112) [reply services/question/src/controller/questions-controller.ts:83]: Agree with don, I think Patch is the one with partial update?

**Review by ys112 — APPROVED** [review]

**Pull request merged**